packages/react-components/source/scss/library/components/_tabs.scss

```diff
@@ -8,30 +11,68 @@
   max-width: 100%;
 }
 
+.rc-tabs-list-container {
+  align-items: flex-end;
+  display: flex;
+}
+
 .rc-tabs-list {
   align-items: flex-end;
+  box-shadow: $bottom-border-shadow;
   display: flex;
-  overflow: visible;
+  min-height: $scroll-button-size + 2; // Accommodate box shadow on focus
+  -ms-overflow-style: none;
+  overflow-x: auto;
+  overflow-y: hidden;
+  position: relative;
+  scroll-behavior: smooth;
+  scrollbar-width: none;
+  width: 100%;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
 }
 
 // Tab
 .rc-tabs-button {
   border-bottom: 0 !important;
   border-radius: $puppet-common-border-radius $puppet-common-border-radius 0 0 !important;
+  box-shadow: $bottom-border-shadow;
   color: $puppet-type-color-base;
   fill: $puppet-type-color-base;
   margin: 0 2px;
-  margin-left: 2px !important;
+  margin-left: 2px !important; // override specific default button + button margin
   padding-left: $puppet-common-spacing-base * 6;
   padding-right: $puppet-common-spacing-base * 6;
   position: relative;
+  white-space: nowrap;
+
+  .rc-tabs-no-scroll & {
+    white-space: pre-wrap;
```

Comment:
Any objections to letting the tab names wrap like this if the width is too narrow and scrolling behavior isn't enabled?